main.py

```
@@ -81,198 +105,331 @@ async def dispatch(self, request: Request, call_next):
 
+                return {
+                    "operation_name": operation_id,
+                    "status": "in_progress",
+                    "user_id": user_id,
+                    "session_id": session_id,
+                    "created_at": operations[operation_id]['created_at']
+                }
+
+            except Exception as e:
+                logger.error(f"Error in GenAI client operation: {str(e)}")
+                raise HTTPException(status_code=500, detail=f"Video generation failed: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)